s3-endpoints.go

```diff
@@ -48,17 +50,19 @@ var awsS3EndpointMap = map[string]string{
 	"us-gov-east-1":  "s3.dualstack.us-gov-east-1.amazonaws.com",
 	"cn-north-1":     "s3.dualstack.cn-north-1.amazonaws.com.cn",
 	"cn-northwest-1": "s3.dualstack.cn-northwest-1.amazonaws.com.cn",
-	"ap-southeast-3": "s3.dualstack.ap-southeast-3.amazonaws.com",
-	"ap-southeast-4": "s3.dualstack.ap-southeast-4.amazonaws.com",
 	"il-central-1":   "s3.dualstack.il-central-1.amazonaws.com",
 }
 
 // getS3Endpoint get Amazon S3 endpoint based on the bucket location.
-func getS3Endpoint(bucketLocation string) (s3Endpoint string) {
+func getS3Endpoint(bucketLocation string, useDualstack bool) (s3Endpoint string) {
 	s3Endpoint, ok := awsS3EndpointMap[bucketLocation]
 	if !ok {
 		// Default to 's3.dualstack.us-east-1.amazonaws.com' endpoint.
 		s3Endpoint = "s3.dualstack.us-east-1.amazonaws.com"
 	}
+	if !useDualstack {
+		// Trim 'dualstack' from the endpoint if disabled.
+		s3Endpoint = "s3." + s3Endpoint[len("s3.dualstack."):]
```

Comment:
```suggestion
		s3Endpoint = "s3." + s3Endpoint[len("s3.dualstack."):]
```

Please have a separate map with other set of valid endpoints. Let's avoid string tricks.

Reply:
I'm concerned that having two separate maps will lead to human-errors in the future. So I updated the changes, keeping a single map, which holds both kinds of endpoints. PHAL